### ISSUE TITLE
Make QuickFolders compatible with Thunderbird 78 ESR

### ISSUE THREAD

**Issue by RealRaven2000** [RealRaven2000/QuickFolders#8]
This issue is specifically for handling all compatibility problems in Thunderbird 70. Feel free to add more bugs *specific to this version* below.

- when right-clicking a tab, its label is removed

**Comment by RealRaven2000** [comment]
Here is a quick-fix for the "removed label" problem; this was caused by some faulty tidy up of old menus in the function addPopupSet:

```
    if (button.firstChild)
      button.removeChild(button.firstChild);
```

I have fixed this by adding && tagname == "menupopup" to the conditional
[QuickFolders-min-4.17pre27.zip](https://github.com/RealRaven2000/QuickFolders/files/3775685/QuickFolders-min-4.17pre27.zip)

**Comment by RealRaven2000** [comment]
Another regression: the search box of the quickMove panel doesn't work. I replaced the (deprecated) textbox with a search-textbox element - for this, I had to split out the tool panel into a separate XUL file. There is still a new issue with loading the reading list which is caused by this fix. test version:

[QuickFolders-wx-4.17pre41.zip](https://github.com/RealRaven2000/QuickFolders/files/3800653/QuickFolders-wx-4.17pre41.zip)

**Comment by RealRaven2000** [comment]
When moving files with quickMove, there is an exception that fails querying the folders ti the nsIRDFResource interface. I was able to remove the references, so the quickMove function works again - fixed version, tested in 72.0a1 Daily (2019-111-01):

[QuickFolders-wx-4.17pre43.zip](https://github.com/RealRaven2000/QuickFolders/files/3801745/QuickFolders-wx-4.17pre43.zip)

**Comment by RealRaven2000** [comment]
I created the new branch ESR78 and have started the conversion process, for real. I estimate about 3-5 weeks.

**Comment by RealRaven2000** [comment]
Just updated manifest,json so that QF actually installs (in Tb78 b4)
Obviously I need to convert the options file, but this is an encouraging start:

![image](https://user-images.githubusercontent.com/6314674/86459732-2a53eb00-bd1f-11ea-9c9f-de1aad4be6a5.png)

**Comment by RealRaven2000** [comment]
Timeline: we intend to be ready by end of September, when 78.2 is auto-rolled out to the Tb68 users base. Please do not upgrade until then.

**Comment by RealRaven2000** [comment]
I had to abandon branch ESR78 because it was too chaotic. Created a new branch here:

[ESR78-method](
https://github.com/RealRaven2000/QuickFolders/tree/ESR78-method)


and now following John Bieling's update guide, here:
https://github.com/thundernest/addon-developer-support/wiki/Using-the-WindowListener-API-to-convert-a-Legacy-Overlay-WebExtension-into-a-MailExtension-for-Thunderbird-78

**Comment by PotatoCarl** [comment]
Would be happy to beta-test for Thunderbird 78.

**Comment by RBZS1947** [comment]
I too would be happy to beta-test for TB78.

**Comment by RealRaven2000** [comment]
I just had some design meeting, and here are some previews for the toolbars:

Current Folder ("Navigation") toolbar:

![image](https://user-images.githubusercontent.com/6314674/89308207-e7609c80-d669-11ea-980a-ce94f7ea2a75.png)

New icons for main toolbar: (Toolbar toggle, create new folder, skip folder)
![image](https://user-images.githubusercontent.com/6314674/89308364-1545e100-d66a-11ea-9d54-71e857510fd6.png)
![image](https://user-images.githubusercontent.com/6314674/89308549-4b836080-d66a-11ea-9b65-c7a3058f60f3.png)
![image](https://user-images.githubusercontent.com/6314674/89308571-52aa6e80-d66a-11ea-96a8-cabcd66ae773.png)

they will of course have "focus colors" to make them even easier to use.

**Comment by opto** [comment]
Hi, I am opto - Klaus. I am updating Quickfolders together with Axel.

We just made a major - if preliminary breakthrough.

As a start, we are putting the new technology into 68. Later, we will transpose that to 78.

Now, as of today, the new technology can in 68:
* drag and drop folders to the toolbar
* if you click that, you will display of the messages in that folder
* context menus on the folders work: you can color, create subfolders - no guarantee all works.

This all works with the new addon technology and can be easily (???) ported to 78. Well, not really easily, but we are promised it will be possible.

If you want to help with testing, that is very welcome:

install a test 68.10 (that is what I am using)
install the test version (Axel will put a link here)

test and report any issues - likely, there will be many, but still, it starts to come together.

enjoy,

Klaus

**Comment by jobisoft** [comment]
RemoveDupes also uses this two-step upgrade process and the developer reported that the final step into TB78 required only little changes. Strategy is looking good!

**Comment by opto** [comment]
and THANKS !! to John for his help

**Comment by RealRaven2000** [comment]
Here is a first prerelease (beta) of the new QuickFolders, based on non-legacy code, It's only tested in Thunderbird 68.10 / 68.11 yet; probably some stuff still missing, but it's useable:

[QuickFolders-wx-5.0pre45.zip](https://github.com/RealRaven2000/QuickFolders/files/5036114/QuickFolders-wx-5.0pre45.zip)

To install, please unpack the contained xpi to hard drive. then pull the xpi file into Add-ons Manager.

Important: This is not tested in Tb78 yet! but it uses methods that (so we are told) are forward compatible. We will see what particular things break once we have completed work on 68 first - for safety we are sticking to the conversion guide here:
https://github.com/thundernest/addon-developer-support/wiki/WindowListener-API:-Getting-Started

**Comment by RealRaven2000** [comment]
> 
> 
> and THANKS !! to John for his help

dito from me! My next plan is to get SmartTemplate4 to the same stage _quickly_...

(Some) users are already screaming for it.

**Comment by RealRaven2000** [comment]
Here is another preview version, based on opto's esr method branch which works partly in Thunderbird 78:

[QuickFolders-wx-5.0pre48.zip](https://github.com/RealRaven2000/QuickFolders/files/5051287/QuickFolders-wx-5.0pre48.zip)

To install, please unpack the contained xpi to hard drive. then pull the xpi file into Add-ons Manager.

What it already does in Thunderbird78: 
- displays your old tabs properly
- click a tab to go to the folder
- drag tabs to change their visual order
- add new tabs by dragging
- context menus of tabs (mostly)

What doesn't work yet (Tb78):
- the options dialog (and anything related to entering licenses) doesn't work in Tb78 yet
- you cannot currently drag mails to the tabs - Mozilla got rid of an important drag and drop interface so we gotta rewrite that part
- there is a bug in the settings menu (clicking on the blue cogwheel doesn;t work) - instead you can use the right-click on the QuickFolders label to display the menu
- any of the rudimentary quickfilter / create filter methology

**Comment by RealRaven2000** [comment]
Here is the next prerelease version. We are having some toruble with the foldertree (I bypassed the code for now) and still no drag and drop, But most dialogs (options / register / categories / change order) work properly now.

This version only works in Thunderbird 78 or higher.

[QuickFolders-wx-5.0pre50.zip](https://github.com/RealRaven2000/QuickFolders/files/5058092/QuickFolders-wx-5.0pre50.zip)

To install, please unpack the contained xpi to hard drive. then pull the xpi file into Add-ons Manager.

**Comment by RealRaven2000** [comment]
And another: I fixed the icon issue in foldertree:

[QuickFolders-wx-5.0pre51.zip](https://github.com/RealRaven2000/QuickFolders/files/5058256/QuickFolders-wx-5.0pre51.zip)

To install, please unpack the contained xpi to hard drive. then pull the xpi file into Add-ons Manager.

- The main thing that doesn't work still is drag+drop. :)

getting closer.

**Comment by RealRaven2000** [comment]
New version - another milestone. Thunderbird 78 only!

[QuickFolders-wx-5.0pre52.zip](https://github.com/RealRaven2000/QuickFolders/files/5064791/QuickFolders-wx-5.0pre52.zip)

To install, please unpack the contained xpi into a hard drive folder. then pull the xpi file into Thunderbird Add-ons Manager.

- we fixed most of the drag+drop issues, still have to do a complete test for this.
- The navigation bar (current folder toolbar) is now in the correct position (on top of the mail preview in 3pane mode)

**Comment by RealRaven2000** [comment]
Next version - prerelease 54:

[QuickFolders-wx-5.0pre54.zip](https://github.com/RealRaven2000/QuickFolders/files/5066322/QuickFolders-wx-5.0pre54.zip)

To install, please unpack the contained xpi into a hard drive folder. then pull the xpi file into Thunderbird Add-ons Manager.

- Fixed the tab reordering issue 
- Fixed the bad optics in the options dialog
- Fixed lightweight themes support in options

To do next:

- make the fieldsets a little friendlier when using the default theme
- backup / restore checkboxes do not remember their values!
- fix dropmarker css when reordering tabs to show where tab will be positioned

**Comment by RealRaven2000** [comment]
Here is the latest update with style related and functional fixes:

[QuickFolders-wx-5.0pre57.zip](https://github.com/RealRaven2000/QuickFolders/files/5075062/QuickFolders-wx-5.0pre57.zip)

To install, please unpack the contained xpi into a hard drive folder. then pull the xpi file into Thunderbird Add-ons Manager.

- made the fieldsets a little friendlier when using the default theme
- improved the layout of color pickers and optimized vertical space in dialog
- backup / restore checkboxes should remember values
- the "Find orphaned Tabs" tool menu item is working now

To do next:

- In the license screen "Get QF Pro"  the fields Email / First Name and Last Name are not populated / cannot be edited.
- the black chevorns and dragover colors do not work - Tb currently ignores all dragover css selectors

**Comment by RealRaven2000** [comment]
Here is the latest update with the new skinnable icons (supports theming colors)

![image](https://user-images.githubusercontent.com/6314674/90287662-37144480-de70-11ea-80b8-360d690df0b9.png)

These icons were created by @Paenglab - many thanks for the beautiful work!

[QuickFolders-wx-5.0pre59.zip](https://github.com/RealRaven2000/QuickFolders/files/5076894/QuickFolders-wx-5.0pre59.zip)

To install, please unpack the contained xpi into a hard drive folder. then pull the xpi file into Thunderbird Add-ons Manager.

Fixed:
  - register / license screen - account selection and editing form fields works now

Known issues:
- the recent folders icon disappears permanently when clicked in navigation bar
- still issues with dragover states not registering
- integrate the new icons into general settings / advanced settings pages.
- style the tabbed icons in the options dialog

**Comment by RealRaven2000** [comment]
After improving some of the icons, here is the latest version:

[QuickFolders-wx-5.0pre62.zip](https://github.com/RealRaven2000/QuickFolders/files/5079372/QuickFolders-wx-5.0pre62.zip)

To install, please unpack the contained xpi into a hard drive folder. then pull the xpi file into Thunderbird Add-ons Manager.

**Comment by jim1506** [comment]
Since all updates after 5.7 there is only one line of quick folder tabs shown in Thunderbird, so I have had to revert always to 5.7. I have quite a few lines as I have several mail accounts.

I have tried finding a setting that say allow multiple lines but cannot.

**Comment by RealRaven2000** [comment]
> Since all updates after 5.7 there is only one line of quick folder tabs shown in Thunderbird, so I have had to revert always to 5.7. I have quite a few lines as I have several mail accounts.
> 
> I have tried finding a setting that say allow multiple lines but cannot.

5.7? you probably mean pre 57 - I see... yeah they need to wrap, that's a regression right there... what's the last versioon where it works?

![image](https://user-images.githubusercontent.com/6314674/90335696-adc75400-dfce-11ea-8a1e-5db8a0694036.png)

**Comment by UdFrederic** [comment]
I tried the "QuickFolders-wx-5.0pre62.zip" in "Thunderbird 78.1.0 (32-Bit)": Drag and drop onto the quick folder icon is possible but the text field which then opens has a red background and does not let me write text/folder name in, so moving of e-mails is not possible. Here how it lools like:

![Bug_QuickFolders_5pre62](https://user-images.githubusercontent.com/16845339/90376020-1cabb800-e076-11ea-986e-193c482f59d5.jpg)

And the version info of Thunderbird:

![Thunderbid 78 1 0](https://user-images.githubusercontent.com/16845339/90376145-4a90fc80-e076-11ea-9f0d-d8c06bc488a7.JPG)

**Comment by RealRaven2000** [comment]
Latest Version:

[QuickFolders-wx-5.0pre63.zip](https://github.com/RealRaven2000/QuickFolders/files/5085808/QuickFolders-wx-5.0pre63.zip)

Fixed:
- multiple lines for the tabs - tabs wrap again
- the recent folder and mail commands icons would disappear after clicking and opening their popup menu

Known Issues:
- no dragover changes. I have filed a [bug with Thunderbird](https://bugzilla.mozilla.org/show_bug.cgi?id=1659232) on this one. I believe it is a problem that Mozilla may have introduced from the Firefox side.

**Comment by waborn427** [comment]
> 
> 
> I tried the "QuickFolders-wx-5.0pre62.zip" in "Thunderbird 78.1.0 (32-Bit)": Drag and drop onto the quick folder icon is possible but the text field which then opens has a red background and does not let me write text/folder name in, so moving of e-mails is not possible. Here how it lools like:
> 
> ![Bug_QuickFolders_5pre62](https://user-images.githubusercontent.com/16845339/90376020-1cabb800-e076-11ea-986e-193c482f59d5.jpg)
> 
> And the version info of Thunderbird:
> 
> ![Thunderbid 78 1 0](https://user-images.githubusercontent.com/16845339/90376495-dc990500-e076-11ea-8032-4a43e0538504.JPG)
> 
> Second observation: without having moved an e-mail to the quick folder icon (so only opening the textbox for to jump to a specific folder) the background is white - but still no text can be typed in:
> 
> ![Bug_QuickFolders_5pre62_Bug2](https://user-images.githubusercontent.com/16845339/90376918-7eb8ed00-e077-11ea-86cd-4969088e6708.jpg)

Same here, dragging mail in learning mode, no dialog opens, no tab creating possible. Win10 2004

**Comment by RealRaven2000** [comment]
Latest version, fixed a few regressions here:

[QuickFolders-wx-5.0pre64.zip](https://github.com/RealRaven2000/QuickFolders/files/5088754/QuickFolders-wx-5.0pre64.zip)

To install, please unpack the contained xpi into a hard drive folder. then pull the xpi file into Thunderbird Add-ons Manager.

Improvements / Fixed
- fixed a regression that broke the text search field for quickMove / quickJump / quickCopy
- fixed disappearing icons (recent folders / mail fodler commands) of commands with popup menus
- more work done by @Paenglab  on icon designs:
![image](https://user-images.githubusercontent.com/6314674/90481349-85a03800-e129-11ea-9c37-8b113d3a4de9.png)
![image](https://user-images.githubusercontent.com/6314674/90481624-ffd0bc80-e129-11ea-8d3f-1cfb24a21c08.png)
![image](https://user-images.githubusercontent.com/6314674/90481673-10813280-e12a-11ea-92e6-465a5ee60f13.png)

**Comment by RealRaven2000** [comment]
I made an official prerelease, for easier installation, please see here:

https://github.com/RealRaven2000/QuickFolders/releases/tag/5.0pre65

- fixed a regression with search boxes (Shift M / Shift J would unintentionally toggle to other mode when used within quickJump text field)
- added fixes for textboxes in Category dialog by @opto

**Comment by waborn427** [comment]
![grafik](https://user-images.githubusercontent.com/69837530/90486603-e471ab80-e141-11ea-84c6-bee8f8be2780.png)
Partially solved
Diaolg box not appearing. (or mayb I misunderstand intended function)

**Comment by UdFrederic** [comment]
> 
> 
> Latest version, fixed a few regressions here:
> 
> [QuickFolders-wx-5.0pre64.zip](https://github.com/RealRaven2000/QuickFolders/files/5088754/QuickFolders-wx-5.0pre64.zip)
> 
> To install, please unpack the contained xpi into a hard drive folder. then pull the xpi file into Thunderbird Add-ons Manager.
> 
> Improvements / Fixed
> 
>     * fixed a regression that broke the text search field for quickMove / quickJump / quickCopy

Yes! You solved it! Now quickMove / quickJump works fine in my config!

**Comment by RealRaven2000** [comment]
> 
> 
> ![grafik](https://user-images.githubusercontent.com/69837530/90486603-e471ab80-e141-11ea-84c6-bee8f8be2780.png)
> Partially solved
> Dialog box not appearing. (or maybe I misunderstand intended function)

Looks like you are doing2 things here - 1 use the quickJump function and 2 use the assistant to create filters (red field behind filter icon).

To be honest this feature ("Create Message Filters") doesn't really work that well in QuickFolders. I had moved the functionality to my [other add-on, quickFilters](https://addons.thunderbird.net/en-US/thunderbird/addon/quickfilters/), where I made it much more advanced. We will hopefully get the conversion of that one finished within the next four weeks...

**Comment by waborn427** [comment]
quickfilters, well, I used that one until now, which did fine, but disappeared as well with THU78. Now I see where my problem is...:)

**Comment by RealRaven2000** [comment]
> 
> 
> quickfilters, well, I used that one until now, which did fine, but disappeared as well with THU78. Now I see where my problem is...:)

Coming soon... First we will deal with SmartTemplate because that has more paying customers. I need to make the money back that I am currently sinking into this "keep up or die" exercise. I think we have already crossed $2,000 this week. Hopefully we will make all 3 commercial Add-ons before Thunderbird pushes the new version to everybody, so transition should be smooth.

**Comment by cube79** [comment]
.Hello. Thank you for quick updates. I'm testing it on Thunderbird 78.1.1. Which was upgraded from version 68. I used separator to divide my QuickFolders, and the current version does it in a weird way, like on the picture below. I cannot delete it unfurtunetaly.
![image](https://user-images.githubusercontent.com/58031508/91004714-419fae00-e5d5-11ea-9fc5-1a0bc4ef0e00.png)

**Comment by RealRaven2000** [comment]
> 
> 
> .Hello. Thank you for quick updates. I'm testing it on Thunderbird 78.1.1. Which was upgraded from version 68. I used separator to divide my QuickFolders, and the current version does it in a weird way, like on the picture below. I cannot delete it unfurtunetaly.

thanks for testing this - I will look into how to fix it. In the meantime, you can delete any separator (including new line) by clicking the _next_ tab (the one after / below the separator) and selecting *Remove Separator* from the *QuickFolders Commands* menu.

![image](https://user-images.githubusercontent.com/6314674/91011273-e9b87580-e5db-11ea-8c75-91c738a1271b.png)

**Comment by cube79** [comment]
> > .Hello. Thank you for quick updates. I'm testing it on Thunderbird 78.1.1. Which was upgraded from version 68. I used separator to divide my QuickFolders, and the current version does it in a weird way, like on the picture below. I cannot delete it unfurtunetaly.
> 
> thanks for testing this - I will look into how to fix it. In the meantime, you can delete any separator (including new line) by clicking the _next_ tab (the one after / below the separator) and selecting _Remove Separator_ from the _QuickFolders Commands_ menu.
> 
> ![image](https://user-images.githubusercontent.com/6314674/91011273-e9b87580-e5db-11ea-8c75-91c738a1271b.png)

I couldn't find this by myself - thank you for pointing it out - worked flawlessly.

**Comment by RealRaven2000** [comment]
Here is an update which fixed the separator issue (extra unwanted line breaks) described previously:

[QuickFolders-wx-5.0pre66.zip](https://github.com/RealRaven2000/QuickFolders/files/5117377/QuickFolders-wx-5.0pre66.zip)

This also includes the latest icon designs from @Paenglab and I must say they look gorgeous! The main thing that remains to be fixed is the drag-over styling, which is going to be a tougher piece of work as the Mozilla specific CSS selector -moz-drag-over has been removed and there is no dragover styling in the html standard.

New icons in the QuickFolders Commands popup menu:
![image](https://user-images.githubusercontent.com/6314674/91038086-15e6ed00-e602-11ea-9700-497431a9830d.png)

New icons in the QuickFolders Options dialog:
![image](https://user-images.githubusercontent.com/6314674/91038133-2d25da80-e602-11ea-9745-95cd1de1fc95.png)

New icons in the commands menu:
![image](https://user-images.githubusercontent.com/6314674/91038241-5e060f80-e602-11ea-9264-22facfbb92fe.png)

**Comment by monnardj** [comment]
Hi, 
I'm currently testing 5.0pre65 on TB 78.2.0, MacOS 10.14.6. Here is what appears when I open a message in new window :

![Untitled](https://user-images.githubusercontent.com/5880250/91530396-4cc34880-e90b-11ea-888c-25fefca766d1.png)
Do you know where this could come from ?

**Comment by RealRaven2000** [comment]
> 
> 
> Hi,
> I'm currently testing 5.0pre65 on TB 78.2.0, MacOS 10.14.6. Here is what appears when I open a message in new window :
> 
> ![Untitled](https://user-images.githubusercontent.com/5880250/91530396-4cc34880-e90b-11ea-888c-25fefca766d1.png)
> Do you know where this could come from ?

Looks to me like the style sheet rules were not loaded, this 
![image](https://user-images.githubusercontent.com/6314674/91638218-a445df00-ea05-11ea-94fc-e6baf55b8611.png)
should look something like this:
![image](https://user-images.githubusercontent.com/6314674/91638228-ac9e1a00-ea05-11ea-93ce-4e31b5e1af19.png)

After [installing prerelease 66](https://github.com/RealRaven2000/QuickFolders/releases/tag/5.0pre66), Can you check whether the main toolbar is displayed? It looks like this:
![image](https://user-images.githubusercontent.com/6314674/91638253-cccdd900-ea05-11ea-8c17-a8798bebc268.png)

if not, please check the JS error console with Tools / Developer Tools / Error Console:

![image](https://user-images.githubusercontent.com/6314674/91638268-fe46a480-ea05-11ea-9add-40fe2f9b7f9b.png)

you disable the addon then clear the output. Then enable QuickFolders and see whether there were any errors / warnings.

**Comment by ThunderbirdMailDE** [comment]
Hi!

5.0pre66 on Tb 78.2.0 on Windows 10 triggers a problem in the main toolbar customization dialog. If your addon is installed (and enabled) and I install my "Allow HTML Temp 6.0" addon, its button isn't shown in the customization dialog, until I disable your addon.

This issue occurs as long as the following code block remains in your qf-messener.js file:

`        
        <toolbarpalette id="MailToolbarPalette">
          <toolbarbutton id="QuickFolders-toolbar-button" 
                class="toolbarbutton-1 chromeclass-toolbar-additional"
                label="&qf.toolbar.quickfolders.toolbar;" 
                tooltiptext="&qf.toolbar.quickfolders.toolbar;"
                oncommand="QuickFolders.Interface.toggleToolbar(this);"
                checked="true"
                />
          <toolbarbutton id="QuickFolders-createfolder" 
                class="toolbarbutton-1 chromeclass-toolbar-additional"
                label="&quickfolders.toolbar.newsubfolder;" 
                tooltiptext="&quickfolders.toolbar.newsubfolder;" 
                oncommand="QuickFolders.Interface.onCreateInstantFolder();"
              />
            <toolbarbutton id="QuickFolders-skipfolder"
                class="toolbarbutton-1 chromeclass-toolbar-additional"
                label="&quickfolders.toolbar.skip;" 
                tooltiptext="&qf.tooltip.skipUnreadFolder;" 
                oncommand="QuickFolders.Interface.onSkipFolder(null);"
              />
        </toolbarpalette>
`

To reproduce this error it's necessary, to dis-/enable my addon evertime, after installing new testversions of your addon. The issue is confirmed by our german moderator "edvoldi" in german support forums.

And BTW the 3 buttons from the codeblock above are not shown at all.

**Comment by monnardj** [comment]
> > Hi,
> > I'm currently testing 5.0pre65 on TB 78.2.0, MacOS 10.14.6. Here is what appears when I open a message in new window :
> > ![Untitled](https://user-images.githubusercontent.com/5880250/91530396-4cc34880-e90b-11ea-888c-25fefca766d1.png)
> > Do you know where this could come from ?
> 
> Looks to me like the style sheet rules were not loaded, this
> ![image](https://user-images.githubusercontent.com/6314674/91638218-a445df00-ea05-11ea-94fc-e6baf55b8611.png)
> should look something like this:
> ![image](https://user-images.githubusercontent.com/6314674/91638228-ac9e1a00-ea05-11ea-93ce-4e31b5e1af19.png)
> 
> After [installing prerelease 66](https://github.com/RealRaven2000/QuickFolders/releases/tag/5.0pre66), Can you check whether the main toolbar is displayed? It looks like this:
> ![image](https://user-images.githubusercontent.com/6314674/91638253-cccdd900-ea05-11ea-8c17-a8798bebc268.png)
> 
> if not, please check the JS error console with Tools / Developer Tools / Error Console:
> 
> ![image](https://user-images.githubusercontent.com/6314674/91638268-fe46a480-ea05-11ea-9add-40fe2f9b7f9b.png)
> 
> you disable the addon then clear the output. Then enable QuickFolders and see whether there were any errors / warnings.

Still doesn't work. 

When I disable and re-enable the extension, the console shows the following warning for line 886 of quickfolders.js :

init: folderEntries
tabMode is not defined

I think the problem might be related to some settings from an older version of QuickFolders. Where/how can I delete these ?

**Comment by RealRaven2000** [comment]
> 
> When I disable and re-enable the extension, the console shows the following warning for line 886 of quickfolders.js :
> 
> init: folderEntries
> tabMode is not defined
> 
> I think the problem might be related to some settings from an older version of QuickFolders. Where/how can I delete these ?

no need - you found a illegal variable scope problem and also you seem to be in some Thunderbird tab that makes this function call fail (or the index I am passing is illegal):
`util.getTabInfoByIndex(tabmail, idx);`

Are you hiding the Thunderbird tabs if only one folder is open?
Here is a quick patch:

[QuickFolders-wx-5.0pre69.zip](https://github.com/RealRaven2000/QuickFolders/files/5149223/QuickFolders-wx-5.0pre69.zip)

I would still be interested in the console log even when it now works. Make sure [x] Debug is switched on in QF advanced settings.

**Comment by monnardj** [comment]
Still not working with version 69.
> Are you hiding the Thunderbird tabs if only one folder is open?

No, I'm not hiding the tabs. But my TB is configured to open messages in a new window, maybe that wasn't clear above. The weird toolbar only appears for messages displayed in their own window, not in the preview pane, and neither when I configure TB to open messages in a new tab.

> I would still be interested in the console log even when it now works. Make sure [x] Debug is switched on in QF advanced settings.

With the new version, there is nothing in the console log with QF Debug switched on.

Possibly Quickfolders's stylesheet should have a different selector for the toolbar for messages displayed in a separate window.

**Comment by ThunderbirdMailDE** [comment]
> Hi!
> 
> 5.0pre66 on Tb 78.2.0 on Windows 10 triggers a problem in the main toolbar customization dialog. If your addon is installed (and enabled) and I install my "Allow HTML Temp 6.0" addon, its button isn't shown in the customization dialog, until I disable your addon.
> 
> This issue occurs as long as the following code block remains in your qf-messener.js file:
> 
> ` <!-- Thunderbird & SeaMonkey --> <toolbarpalette id="MailToolbarPalette"> <toolbarbutton id="QuickFolders-toolbar-button" class="toolbarbutton-1 chromeclass-toolbar-additional" label="&qf.toolbar.quickfolders.toolbar;" tooltiptext="&qf.toolbar.quickfolders.toolbar;" oncommand="QuickFolders.Interface.toggleToolbar(this);" checked="true" /> <toolbarbutton id="QuickFolders-createfolder" class="toolbarbutton-1 chromeclass-toolbar-additional" label="&quickfolders.toolbar.newsubfolder;" tooltiptext="&quickfolders.toolbar.newsubfolder;" oncommand="QuickFolders.Interface.onCreateInstantFolder();" /> <toolbarbutton id="QuickFolders-skipfolder" class="toolbarbutton-1 chromeclass-toolbar-additional" label="&quickfolders.toolbar.skip;" tooltiptext="&qf.tooltip.skipUnreadFolder;" oncommand="QuickFolders.Interface.onSkipFolder(null);" /> </toolbarpalette>`
> 
> To reproduce this error it's necessary, to dis-/enable my addon evertime, after installing new testversions of your addon. The issue is confirmed by our german moderator "edvoldi" in german support forums.
> 
> And BTW the 3 buttons from the codeblock above are not shown at all.


Problem found:
Your add-on uses the outdated WindowsListener-API in version 1.14 from John Bieling. Please, update the API to version 1.20. Then the interference with my Add-on is gone.

**Comment by ThunderbirdMailDE** [comment]
> Your add-on uses the outdated WindowsListener-API in version 1.14 from John Bieling. Please, update the API to version 1.20. Then the interference with my Add-on is gone.

And one more nice thing for you: Using the API 1.20 reveals your Buttons out of nirvana.

**Comment by jobisoft** [comment]
@RealRaven2000 : I pushed the req. updates to ESR78-method-opto. You only have to release a new version.

**Comment by RealRaven2000** [comment]
> 
> Still not working with version 69.
> 
> > Are you hiding the Thunderbird tabs if only one folder is open?
> 
> No, I'm not hiding the tabs. But my TB is configured to open messages in a new window, maybe that wasn't clear above. The weird toolbar only appears for messages displayed in their own window, not in the preview pane, and neither when I configure TB to open messages in a new tab.

Hmm I missed that - here is a fixed version:
https://github.com/RealRaven2000/QuickFolders/releases/tag/5.0pre71

**Comment by RealRaven2000** [comment]
> 
> 
> > Still not working with version 69.
> > > Are you hiding the Thunderbird tabs if only one folder is open?
> > 
> > 
> > No, I'm not hiding the tabs. But my TB is configured to open messages in a new window, maybe that wasn't clear above. The weird toolbar only appears for messages displayed in their own window, not in the preview pane, and neither when I configure TB to open messages in a new tab.
> 
> Hmm I missed that - here is a fixed version:
> https://github.com/RealRaven2000/QuickFolders/releases/tag/5.0pre71



> 
> 
> @RealRaven2000 : I pushed the req. updates to ESR78-method-opto. You only have to release a new version.

Thanks John, I just built a new version with that.

**Comment by monnardj** [comment]
> Hmm I missed that - here is a fixed version:
> https://github.com/RealRaven2000/QuickFolders/releases/tag/5.0pre71

Ok, now the toolbar displays correctly ! But I can't get rid of it (which was actually my original goal : in the previous version of QF, I didn't use it) : 
- the close button (X) doesn't work
- the settings have no effect whatsoever (even for the QF toolbar in the TB main window, which is always displayed)
<img width="834" alt="qf" src="https://user-images.githubusercontent.com/5880250/91824714-7e098480-ec3b-11ea-8bce-134e4379e858.png">

**Comment by RealRaven2000** [comment]
You found the right panel for disabling:
![image](https://user-images.githubusercontent.com/6314674/91833142-37695980-ec3e-11ea-90e0-3c4866a152b4.png)
but obviously something is wrong probably missing a script somewhere, should be easy enough to fix.

**Comment by pgnd** [comment]
@RealRaven2000 

Apparently Mozilla just pushed updates to 78.2.2.

I've a number of user that accepted the updates.  Of course, breaking 'everything' ...

Are there compatible updates of QF + your apps -- even in beta -- yet?

If so, can you pls point?

If not, I'll scramble to rollback to T68.

**Comment by RealRaven2000** [comment]
> Are there compatible updates of QF + your apps -- even in beta -- yet?
> 
> If so, can you pls point?
> 
> If not, I'll scramble to rollback to T68.

yes there is:

the freshest prerelease is [Release 5.0 pre 71](https://github.com/RealRaven2000/QuickFolders/releases/tag/5.0pre71).

**Comment by RealRaven2000** [comment]
Latest Version:

Just Released [Version 5.0pre80](https://github.com/RealRaven2000/QuickFolders/releases/tag/5.0pre80)

- Fixed layout in update / install popups 
- Fixed issue #55 
- Made new shortcut (F7) for toggling the folder pane.

**Comment by Vacoas** [comment]
Many thanks for the great work in making QuickFolders Thunderbird-78 compatible.
I have encountered a small regression moving from pre70 to pre81.
I use letter 'm' for quickMove in the Premium functionality pane. This does not work anymore : it just marks the mail as unread.

**Comment by RealRaven2000** [comment]
>  I use letter 'm' for quickMove in the Premium functionality pane. This does not work anymore : it just marks the mail as unread.

**Comment by RealRaven2000** [comment]
>  I use letter 'm' for quickMove in the Premium functionality pane. This does not work anymore : it just marks the mail as unread.

**Comment by RealRaven2000** [comment]
>  I use letter 'm' for quickMove in the Premium functionality pane. This does not work anymore : it just marks the mail as unread.

are you sure it is not Shift+M?

**Comment by RealRaven2000** [comment]
>  I use letter 'm' for quickMove in the Premium functionality pane. This does not work anymore : it just marks the mail as unread.

are you sure it is not Shift+M?

**Comment by RealRaven2000** [comment]
>  I use letter 'm' for quickMove in the Premium functionality pane. This does not work anymore : it just marks the mail as unread.

are you sure it is not Shift+M?

**Comment by RealRaven2000** [comment]
>  I use letter 'm' for quickMove in the Premium functionality pane. This does not work anymore : it just marks the mail as unread.

are you sure it is not Shift+M?

**Comment by RealRaven2000** [comment]
>  I use letter 'm' for quickMove in the Premium functionality pane. This does not work anymore : it just marks the mail as unread.

are you sure it is not Shift+M?

**Comment by RealRaven2000** [comment]
>  I use letter 'm' for quickMove in the Premium functionality pane. This does not work anymore : it just marks the mail as unread.

are you sure it is not Shift+M?

**Comment by Vacoas** [comment]
It is just 'm'. I have reverted to pre70 and it works.

![quickMove](https://user-images.githubusercontent.com/45902444/93434233-28cea380-f8c8-11ea-826c-c6d45ec7e65e.jpg)

**Comment by RealRaven2000** [comment]
Some of the shortcuts are probably taken by Thunderbird themselves (I know for a fact that M toggles unread status on my Tb68, so I am surprised it doesn't do that on yours). Anyway if you want a shortcut without Shift you have to look for a different letter. Thunderbird probably won't allow us to overwrite the built in keyboard shortcuts.

**Comment by Vacoas** [comment]
Many thanks for replying quickly. You are right. It is indeed surprising 'm' has worked so far. I'll choose another lower case letter after checking it is not used by TB.

**Comment by Vacoas** [comment]
Hello again, 
I am afraid that the shortcuts for quickMove and quickJump do not work after 5.0pre71. It dos not matter which letter and which case is used.
Before testing a pre-version, I have removed the previous one but I have not gone as far as removing the Thunderbird parameters for quickfolders. 
From the testing done, I believe 5.0pre71 reads and uses 'extensions.quickfolders.quick..." configuration items although it will only update the 'qfpp' configuration items when the key configs are changed.
5.0pre84 updates the 'qfpp' configuration items but appears not to use them.

pre71

![pre71 config](https://user-images.githubusercontent.com/45902444/93572731-8af5da00-f996-11ea-9fe4-7903d57f6e6b.jpg)

pre84

![pre84 config qfpp](https://user-images.githubusercontent.com/45902444/93572808-a5c84e80-f996-11ea-80c1-41a3c807d3d6.jpg)

I can provide more information if needed.

Regards.

**Comment by RealRaven2000** [comment]
> 
> 
> Hello again,
> I am afraid that the shortcuts for quickMove and quickJump do not work after 5.0pre71. It dos not matter which letter and which case is used.
> Before testing a pre-version, I have removed the previous one but I have not gone as far as removing the Thunderbird parameters for quickfolders.
> From the testing done, I believe 5.0pre71 reads and uses 'extensions.quickfolders.quick..." configuration items although it will only update the 'qfpp' configuration items when the key configs are changed.

It sounds like a regression - @opto pointed out to me that the use of keyCode was deprecated so he changed it to "charCode" in the last patch, I am going to test that now...

**Comment by jobisoft** [comment]
Actually both are deprecated. Use

https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/code

instead.

**Comment by Vacoas** [comment]
quickMove and quickJump issue fixed in 5.0pre87.
Many thanks for the great work.

**Comment by RealRaven2000** [comment]
> 
> 
> Hello again,
> I am afraid that the shortcuts for quickMove and quickJump do not work after 5.0pre71. It dos not matter which letter and which case is used.
> Before testing a pre-version, I have removed the previous one but I have not gone as far as removing the Thunderbird parameters for quickfolders.
> From the testing done, I believe 5.0pre71 reads and uses 'extensions.quickfolders.quick..." configuration items although it will only update the 'qfpp' configuration items when the key configs are changed.
> 5.0pre84 updates the 'qfpp' configuration items but appears not to use them.
> 
That's correct, none of the settings starting with "qf"  should actually be in the config database (I only use these internally to categorize different setting groups), so there are definitely some major fixes necessary still. All user specific preferences should start with 

`extensions.quickfolders`

I am hoping to get some assistance from @jobisoft early next week to fix this issue. Maybe we will switch to local storage for good, but I want to be able to still see changes (instantApply) visually in the toolbar while the options dialog is open.

**Comment by nandudb** [comment]
i installed the quickfolders from below url, but I don't see it picked up my previous configurations and folder colors as it shows  nothing. Is there way to copy or backup the configuration from older thunderbird quickfolders extension and restore it new thunderbird(78.2)? I don't want to manually create folders/colors.

https://github.com/RealRaven2000/QuickFolders/releases/tag/5.0pre71

**Comment by RealRaven2000** [comment]
Here is a new pre-release. The preferences system was borked, we fixed it with @jobisoft today:

https://github.com/RealRaven2000/QuickFolders/releases/tag/5.0pre90

**Comment by RealRaven2000** [comment]
> 
> 
> i installed the quickfolders from below url, but I don't see it picked up my previous configurations and folder colors as it shows nothing. Is there way to copy or backup the configuration from older thunderbird quickfolders extension and restore it new Thunderbird(78.2)? I don't want to manually create folders/colors.

yes there is - it's on the QuickFolders Pro tab:
![image](https://user-images.githubusercontent.com/6314674/93859494-965a4580-fcb5-11ea-9fa1-311e4da52a74.png)



If it is on the same PC, you can also use these two buttons on the advanced tab which use the Clipboard for only the folder configuration (all tabs):
![image](https://user-images.githubusercontent.com/6314674/93859471-8fcbce00-fcb5-11ea-9372-505969bbe22b.png)

**Comment by RealRaven2000** [comment]
There is a new prerelase available, please see:

https://github.com/RealRaven2000/QuickFolders/releases/tag/5.0pre100

**Comment by Vacoas** [comment]
I have a problem 5.0.pre100 which did not occur with 5.0.pre92.
All tabs positions and colours are displayed as before.
It's just the colours that have disappeared from the QuickFolders settings. 
I tried to reload in pre100 the settings I had saved in pre92 but this made no difference.
It's probably just a display problem in the 'Bling my tabs' tab.
![pre92](https://user-images.githubusercontent.com/45902444/95983459-ed20ee00-0e21-11eb-9b09-c3c7336810cc.jpg)
![pre100](https://user-images.githubusercontent.com/45902444/95983474-f4e09280-0e21-11eb-95b8-c45e7378a99a.jpg)

**Comment by RealRaven2000** [comment]
> 
> 
> I have a problem 5.0.pre100 which did not occur with 5.0.pre92.
> All tabs positions and colours are displayed as before.
> It's just the colours that have disappeared from the QuickFolders settings.
> I tried to reload in pre100 the settings I had saved in pre92 but this made no difference.
> It's probably just a display problem in the 'Bling my tabs' tab.

It looks like something in the code crashes when the options dialog is opened (I can see even your Pro tab isn't initialized properly).

Can you create a debug log (just open the JavaScript error console [which you can open with Ctrl+Shift+J] before you open options) and copy out the results using the command "_export visible messages to clipboard_" after opening the options dialog?
![image](https://user-images.githubusercontent.com/6314674/95988113-4daf2b80-0e20-11eb-90b0-acf4b3926e51.png)

Before you test, delete the contents of the error console using the button on the top left:
![image](https://user-images.githubusercontent.com/6314674/95988243-86e79b80-0e20-11eb-9fd9-5d79a8f90cda.png)

then open options and capture the log - if you want you can send me the log directly off list (to the support email address given on the options screen / support website) for keeping privacy.

**Comment by RealRaven2000** [comment]
After @Vacoas sent me en error log I found the problem can be reproduced by opening the QuickFolders options dialog through the Add-ons / Tools menu. Some code that is designed to preselect a tab (and possibly hide others) in this dialog fails because of the window arguments parameter being passed having an unexpected length; this also leads to the license validation misfiring. It will be fixed in the next pre-release.

**Comment by RealRaven2000** [comment]
Discovered an issue with the category manger not loading due to missing namespace for the new html input controls. I have already fixed this and am going to add this to the next prerelease. In the meantime, the very latest beta version can be found here:

https://github.com/RealRaven2000/QuickFolders/releases/tag/5.0pre116

- improved donation link
- fully localized splash screens (18 languages)
- using new Add-on Icon designed by @Paenglab

**Comment by nobswolf** [comment]
pre116 works nicely here. It found all folders I used in the old version. Environment is Debian 10.6 and TB 78.4.0 64 Bit.

**Comment by oldmoldy** [comment]
My Thunderbird upgraded itself to 78.4.0 in Windows 10 and the pre116 release solved my problems.  I downloaded the **.xpi** file and then went into TBird Add-on Manager.  Using drag and drop I was able to install the file successfully.  Thanks.

**Comment by RealRaven2000** [comment]
Latest prerelease solving the broken category dialog mentioned in issue #68:

https://github.com/RealRaven2000/QuickFolders/releases/tag/5.0pre119

**Comment by phit42** [comment]
I believe 78.4 was not optional anymore but rolled out to a lot of users, maybe all?

**Comment by annepiez** [comment]
just want to say a big thank you - so helpful!

**Comment by RealRaven2000** [comment]
Released 5.0 for Thunderbird 78 - 17/11/2020 (currently waiting for approval on ATN)

if you don't have a pro license get the Black Friday bargain 19/Nov - 29/November! - or get one for a friend!

**Comment by LinuxOnTheDesktop** [comment]
@RealRaven2000 

That's good and indeed I received a pop-up in Thunderbird and I renewed my license.

However, your [webpage](https://quickfolders.org/index.html) and [version history page](https://quickfolders.org/version.html) do not mention version 5.0. I imagine that that is so because of the approval that you mention you are waiting on. Still, it would be good to do all those things in one go, such that users do not get automatically upgraded to 5.0 until there is a release announcement on the website. That said, perhaps I got upgraded only because I was using a beta.

**Comment by RealRaven2000** [comment]
> @RealRaven2000
> 
> That's good and indeed I received a pop-up in Thunderbird and I renewed my license.
> 
> However, your [webpage](https://quickfolders.org/index.html) and [version history page](https://quickfolders.org/version.html) do not mention version 5.0. I imagine that that is so because of the approval that you mention you are waiting on. Still, it would be good to do all those things in one go, such that users do not get automatically upgraded to 5.0 until there is a release announcement on the website. That said, perhaps I got upgraded only because I was using a beta.

Well they don't do it on top of the page, but new versions are jumped to in a thunderbird tab via a version tag, see:
https://quickfolders.org/version.html#5.0

you are automatically updated from any previous version number (as long as the Add-on is compatible with your current version of Thunderbird). according to the version comparison algorithms: 5.0 > 5.0pre120 > 5.0pre56 > 4.12

**Comment by RealRaven2000** [comment]
Closing issue, with version [5.1](https://github.com/RealRaven2000/QuickFolders/releases/tag/5.1) released on  we are fully compatible with Thunderbird 78
